packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -848,9 +848,29 @@ export class SDKClient {
     callerName: string,
     interactingEntity: string,
     originalCallerAddress: string,
+    currentNetworkExchangeRateInCents: number,
   ): Promise<FileId | null> {
     const hexedCallData = Buffer.from(callData).toString('hex');
 
+    const estimatedTxFee = Utils.estimateFileTransactionsFee(
```

Comment:
Why are we adding this here and what is the benefit of this new function estimateFileTransactionsFee? There seems to be some confusion, as the original intent of this PR was to swap out the old service for the new one. Any insights on how this aligns with the scope of the change would be helpful. Thank you!

Reply:
> Why are we adding this here, and what is the benefit of the new function `estimateFileTransactionsFee()`?

The `estimateFileTransactionsFee()` function is designed to calculate the total transaction fees for file creation and file append transactions based on the size of the call data. This estimated fee is then used for the preemptive rate-limiting logic, as outlined in the design document [here](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/docs/design/hbar-limiter.md#early-detection-and-prevention-preemptive-rate-limit).

> There seems to be some confusion, as the original intent of this PR was to swap out the old service for the new one.

Yes, I understand the concern. However, replacing the `HbarLimit` module with the new `HbarLimitService` class is not as simple as swapping one for the other. There are key differences between the two classes that require additional logic to ensure compatibility. In this specific case, in the former `HbarLimit` module, the preemptive rate-limiting logic was implemented in a separate method, whereas in the new `HbarLimitService`, it has been merged into the `shouldLimit()` function. 

In the previous implementation, the estimated transaction fee was calculated within the corresponding method for preemptive rate-limiting. In the new implementation, the logic has been integrated with the `shouldLimit()` function, which is why the estimated transaction fee calculation is externalized here. This ensures it can still be used in preemptive rate-limiting for file transactions. The rationale for placing it before `fileCreate` and `fileAppend` is explained in [this comment](https://github.com/hashgraph/hedera-json-rpc-relay/pull/3024#discussion_r1781555205).